pgvectorscale/src/access_method/plain_storage.rs

```diff
@@ -252,28 +259,31 @@ impl Storage for PlainStorage<'_> {
         stats: &mut S,
     ) {
         let rn = unsafe { PlainNode::read(self.index, neighbors_of, stats) };
-        //get neighbors copy before givining ownership of rn to the distance state
+        // Copy neighbors before giving ownership of `rn`` to the distance state
         let neighbors: Vec<_> = rn.get_archived_node().iter_neighbors().collect();
         let dist_state = unsafe { IndexFullDistanceMeasure::with_readable_node(self, rn) };
-        for n in neighbors {
+        for n in neighbors.into_iter() {
+            // TODO: we are reading node twice
             let dist = unsafe { dist_state.get_distance(n, stats) };
             result.push(NeighborWithDistance::new(
                 n,
                 DistanceWithTieBreak::new(dist, neighbors_of, n),
+                None,
             ))
         }
     }
 
     /* get_lsn and visit_lsn are different because the distance
     comparisons for SBQ get the vector from different places */
-    fn create_lsn_for_init_id(
+    fn create_lsn_for_start_node(
         &self,
         lsr: &mut ListSearchResult<Self::QueryDistanceMeasure, Self::LSNPrivateData>,
         index_pointer: ItemPointer,
         gns: &GraphNeighborStore,
-    ) -> ListSearchNeighbor<Self::LSNPrivateData> {
+    ) -> Option<ListSearchNeighbor<Self::LSNPrivateData>> {
         if !lsr.prepare_insert(index_pointer) {
-            panic!("should not have had an init id already inserted");
+            info!("Node already processed, skipping");
```

Comment:
wait why do we need to change this in plain storage where we don't need to support labels?